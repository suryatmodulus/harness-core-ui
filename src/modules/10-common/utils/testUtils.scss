/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
