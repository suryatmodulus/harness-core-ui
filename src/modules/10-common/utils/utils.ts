/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
