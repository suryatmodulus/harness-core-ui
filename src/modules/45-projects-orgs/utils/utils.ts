/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { SelectOption } from '@wings-software/uicore'
import { getScopeFromDTO, ScopedObjectDTO } from '@common/components/EntityReference/EntityReference'
import { Scope } from '@common/interfaces/SecretsInterface'
import routes from '@common/RouteDefinitions'
import type { StringKeys, StringsMap } from 'framework/strings/StringsContext'
import { ModuleName } from 'framework/types/ModuleName'

interface RoleOption extends SelectOption {
  managed: boolean
}

interface ScopedDTO {
  accountId: string
  orgIdentifier?: string
  projectIdentifier?: string
}

export const getModuleTitle = (module: ModuleName): keyof StringsMap => {
  switch (module) {
    case ModuleName.CV:
      return 'projectsOrgs.purposeList.change'
    case ModuleName.CE:
      return 'common.purpose.ce.cloudCost'
    case ModuleName.CF:
      return 'common.purpose.cf.feature'
    case ModuleName.CD:
    case ModuleName.CI:
    default:
      return 'projectsOrgs.purposeList.continuous'
  }
}

export const getModulePurpose = (module: ModuleName): keyof StringsMap | undefined => {
  switch (module) {
    case ModuleName.CD:
      return 'common.purpose.cd.delivery'
    case ModuleName.CV:
      return 'common.purpose.cv.intelligence'
    case ModuleName.CI:
      return 'common.purpose.ci.integration'
    case ModuleName.CE:
      return 'common.purpose.ce.management'
    case ModuleName.CF:
      return 'common.purpose.cf.flags'
  }
}
export const getModuleDescriptionsForModuleSelectionDialog = (module: ModuleName): keyof StringsMap | undefined => {
  switch (module) {
    case ModuleName.CD:
      return 'common.selectAVersion.description'
    case ModuleName.CV:
      return 'common.purpose.cv.moduleSelectionSubHeading'
    case ModuleName.CI:
      return 'common.purpose.ci.descriptionOnly'
    case ModuleName.CE:
      return 'common.purpose.ce.moduleSelectionSubHeading'
    case ModuleName.CF:
      return 'common.purpose.cf.moduleSelectionSubHeading'
  }
}
export const getModuleDescription = (module: ModuleName): StringKeys => {
  switch (module) {
    case ModuleName.CD:
      return 'projectsOrgs.purposeList.descriptionCD'
    case ModuleName.CV:
      return 'projectsOrgs.purposeList.descriptionCV'
    case ModuleName.CI:
      return 'projectsOrgs.purposeList.descriptionCI'
    case ModuleName.CE:
      return 'projectsOrgs.purposeList.descriptionCE'
    case ModuleName.CF:
      return 'projectsOrgs.purposeList.descriptionCF'
  }
  return 'projectsOrgs.blank'
}

export const getDefaultRole = (scope: ScopedObjectDTO, getString: (key: keyof StringsMap) => string): RoleOption => {
  if (getScopeFromDTO(scope) === Scope.PROJECT) {
    return { label: getString('common.projectViewer'), value: '_project_viewer', managed: true }
  }
  if (getScopeFromDTO(scope) === Scope.ORG) {
    return {
      label: getString('common.orgViewer'),
      value: '_organization_viewer',
      managed: true
    }
  }
  return { label: getString('common.accViewer'), value: '_account_viewer', managed: true }
}

export const getDetailsUrl = ({ accountId, orgIdentifier, projectIdentifier }: ScopedDTO): string => {
  if (projectIdentifier && orgIdentifier) {
    return `${window.location.href.split('#')[0]}#${routes.toProjectDetails({
      accountId,
      orgIdentifier,
      projectIdentifier
    })}`
  }
  if (orgIdentifier) {
    return `${window.location.href.split('#')[0]}#${routes.toOrganizationDetails({ accountId, orgIdentifier })}`
  }
  return ''
}
export const getModuleFullLengthTitle = (module: ModuleName): keyof StringsMap => {
  switch (module) {
    case ModuleName.CV:
      return 'common.purpose.cv.continuous'
    case ModuleName.CE:
      return 'common.purpose.ce.continuous'
    case ModuleName.CF:
      return 'common.purpose.cf.continuous'
    case ModuleName.CI:
      return 'common.purpose.ci.continuous'
    default:
      return 'common.purpose.cd.continuous'
  }
}
