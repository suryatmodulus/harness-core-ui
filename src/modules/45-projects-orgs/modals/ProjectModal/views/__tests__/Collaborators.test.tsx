/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 */

import React from 'react'

import { fireEvent, getByText, render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import type { Project } from 'services/cd-ng'
import { findPopoverContainer, TestWrapper } from '@common/utils/testUtils'
import Collaborators from '../Collaborators'
import { userMockData } from './UserMockData'
import { invitesMockData, roleMockData } from './InvitesMockData'

const project: Project = {
  orgIdentifier: 'testOrg',
  identifier: 'test',
  name: 'test',
  color: '#e6b800',
  modules: ['CD'],
  description: 'test',
  tags: { tag1: '', tag2: 'tag3' }
}

const response = {
  status: 'SUCCESS',
  data: true,
  metaData: {},
  correlationId: ''
}

jest.mock('@common/hooks', () => ({
  ...(jest.requireActual('@common/hooks') as any),
  useMutateAsGet: jest.fn().mockImplementation(() => {
    return { data: userMockData, refetch: jest.fn(), error: null, loading: false }
  })
}))

jest.mock('services/cd-ng', () => ({
  useGetInvites: jest.fn().mockImplementation(() => ({ data: invitesMockData, loading: false, refetch: jest.fn() })),
  useAddUsers: jest.fn().mockImplementation(() => ({ mutate: () => Promise.resolve(response) })),
  useDeleteInvite: jest.fn().mockImplementation(() => ({ mutate: () => Promise.resolve(response) })),
  useUpdateInvite: jest.fn().mockImplementation(() => ({ mutate: () => Promise.resolve(response) }))
}))

jest.mock('services/rbac', () => ({
  useGetRoleList: jest.fn().mockImplementation(() => ({ data: roleMockData, loading: false, refetch: jest.fn() }))
}))

describe('Collaborators test', () => {
  test('Render Collaborators ', async () => {
    const { container } = render(
      <TestWrapper path="/account/:accountId" pathParams={{ accountId: 'testAcc' }}>
        <Collaborators projectIdentifier={project.identifier} orgIdentifier={project.orgIdentifier} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
    const role = getByText(container, 'Project Viewer')
    fireEvent.click(role!)
    const popover = findPopoverContainer()
    const opt = getByText(popover as HTMLElement, 'Project Admin')
    fireEvent.click(opt!)
    await act(async () => {
      fireEvent.change(container.querySelector("input[name='projectsOrgs.collaborator']")!, {
        target: { value: 'example5@email.com' }
      })
      fireEvent.click(container.querySelector('button[type="submit"]')!)
    })
    await act(async () => {
      const refresh = container.querySelectorAll('[icon="repeat"]')[0]
      fireEvent.click(refresh)
    })
    await act(async () => {
      const deleteInvite = container.querySelectorAll('[icon="trash"]')[0]
      fireEvent.click(deleteInvite)
    })
  })
})
