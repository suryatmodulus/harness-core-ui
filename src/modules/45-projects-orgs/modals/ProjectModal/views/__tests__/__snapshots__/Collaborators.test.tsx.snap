// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collaborators test Render Collaborators  1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      class="FormikForm--main"
    >
      <div>
        <div
          class="StyledProps--font StyledProps--main collaboratorForm"
        >
          <h3
            class="StyledProps--font StyledProps--font-h3 StyledProps--main StyledProps--font-variation-h3 StyledProps--padding-bottom-xxlarge"
            level="3"
          >
            projectsOrgs.invite
          </h3>
          <label
            class="Label--main"
          >
            projectsOrgs.urlMessageProject
          </label>
          <div
            class="TextInput--main"
          >
            <div
              class="bp3-input-group bp3-disabled"
            >
              <input
                class="bp3-input"
                disabled=""
                placeholder="http://localhost/#/account/testAcc/home/orgs/testOrg/projects/test/details"
                style="padding-right: 0px;"
                type="text"
              />
              <span
                class="bp3-input-action"
              >
                <div
                  class="main copy"
                >
                  <div
                    class="text"
                  />
                </div>
              </span>
            </div>
          </div>
          <div
            class="Layout--horizontal StyledProps--main align input StyledProps--padding-top-medium StyledProps--flex"
          >
            <div
              class="Layout--horizontal StyledProps--main"
              style="width: 50%;"
            >
              <label
                class="Label--main"
              >
                projectsOrgs.inviteCollab
              </label>
            </div>
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-end StyledProps--padding-right-medium"
              style="width: 50%;"
            >
              <label
                class="Label--main"
              >
                <div
                  class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--flex"
                >
                  projectsOrgs.roleLabel
                  <div
                    class="bp3-popover-wrapper DropDown--main"
                  >
                    <div
                      class="bp3-popover-target"
                    >
                      <div
                        class=""
                      >
                        <div
                          class="Layout--horizontal StyledProps--main DropDown--dropdownButton StyledProps--flex"
                          data-testid="dropdown-button"
                          style="width: 160px;"
                        >
                          <div
                            class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                          >
                            <p
                              class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                              data-testid="dropdown-value"
                              style="--text-line-clamp: 1;"
                            >
                              Project Viewer
                            </p>
                          </div>
                          <div
                            class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                          >
                            <span
                              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                              data-icon="main-chevron-down"
                            >
                              <svg
                                height="8"
                                viewBox="0 0 14 8"
                                width="8"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                                  fill="#4d5969"
                                />
                              </svg>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </label>
            </div>
          </div>
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
          >
            <div
              class="bp3-form-group input"
              data-id="projectsOrgs.collaborator-0"
            >
              <div
                class="bp3-form-content"
              >
                <div
                  class="bp3-popover-wrapper MultiSelect--main bp3-fill"
                >
                  <div
                    class="bp3-popover-target"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="bp3-input bp3-tag-input bp3-multi-select"
                      >
                        <div
                          class="bp3-tag-input-values"
                        >
                          <input
                            autocomplete="off"
                            class="bp3-input-ghost bp3-multi-select-tag-input-input"
                            name="projectsOrgs.collaborator"
                            placeholder="- Select -"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main StyledProps--inline Button--with-current-color Button--variation Button--variation-primary"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="bp3-button-text"
              >
                add
              </span>
            </button>
          </div>
          <div
            class="Layout--vertical StyledProps--main StyledProps--padding-top-medium StyledProps--padding-bottom-xxxlarge"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--padding-bottom-small"
            >
              projectsOrgs.pendingUsers
            </p>
            <div
              class="StyledProps--font StyledProps--main pendingList"
            >
              <div
                class="StyledProps--font StyledProps--main invites StyledProps--padding-left-xsmall StyledProps--padding-top-medium StyledProps--padding-bottom-medium"
              >
                <div
                  class="Layout--horizontal StyledProps--main"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main align pendingUser"
                    style="width: 60%;"
                  >
                    <div
                      class="Avatar--Avatar Avatar--contentStyle"
                    >
                      <span
                        class="bp3-popover-wrapper Avatar--avatarPopOver"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <div
                            class="Avatar--AvatarInner"
                            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
                            tabindex="0"
                          >
                            E
                          </div>
                        </span>
                      </span>
                    </div>
                    <div
                      class="Layout--vertical StyledProps--main StyledProps--padding-left-small"
                    >
                      <div
                        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                      >
                        <p
                          class="StyledProps--font Text--lineclamp StyledProps--main name StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                          style="--text-line-clamp: 1;"
                        >
                          example4
                        </p>
                        <p
                          class="StyledProps--font StyledProps--main colorBar pending StyledProps--font-size-xsmall StyledProps--font-weight-bold StyledProps--color StyledProps--color-primary7"
                        >
                          projectsOrgs.pendingInvitation
                        </p>
                      </div>
                      <p
                        class="StyledProps--font Text--lineclamp StyledProps--main email"
                        style="--text-line-clamp: 1;"
                      >
                        example4@email.com
                      </p>
                      <div
                        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                      >
                        <p
                          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                        >
                          projectsOrgs.roleAssigned
                        </p>
                        <p
                          class="StyledProps--font Text--lineclamp StyledProps--main role StyledProps--font-variation-tiny"
                          style="--text-line-clamp: 1;"
                        >
                          role 1
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Layout--horizontal StyledProps--main align toEnd StyledProps--padding-right-medium"
                    style="width: 40%;"
                  >
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--inline Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-repeat StyledProps--main"
                            icon="repeat"
                          >
                            <svg
                              data-icon="repeat"
                              height="16"
                              viewBox="0 0 16 16"
                              width="16"
                            >
                              <desc>
                                repeat
                              </desc>
                              <path
                                d="M10 5c0 .55.45 1 1 1h4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1s-1 .45-1 1v1.74A7.95 7.95 0 008 0C3.58 0 0 3.58 0 8c0 4.06 3.02 7.4 6.94 7.92.02 0 .04.01.06.01.33.04.66.07 1 .07 4.42 0 8-3.58 8-8 0-.55-.45-1-1-1s-1 .45-1 1c0 3.31-2.69 6-6 6-.71 0-1.37-.15-2-.38v.01C3.67 12.81 2 10.61 2 8c0-3.31 2.69-6 6-6 1.77 0 3.36.78 4.46 2H11c-.55 0-1 .45-1 1z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--inline Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-trash StyledProps--main"
                            icon="trash"
                          >
                            <svg
                              data-icon="trash"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <desc>
                                trash
                              </desc>
                              <path
                                d="M17 1h-5c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1H3c-.55 0-1 .45-1 1v1h16V2c0-.55-.45-1-1-1zm.5 3h-15c-.28 0-.5.22-.5.5s.22.5.5.5H3v14c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V5h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zM7 16c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8zm4 0c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8zm4 0c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="StyledProps--font StyledProps--main invites StyledProps--padding-left-xsmall StyledProps--padding-top-medium StyledProps--padding-bottom-medium"
              >
                <div
                  class="Layout--horizontal StyledProps--main"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main align pendingUser"
                    style="width: 60%;"
                  >
                    <div
                      class="Avatar--Avatar Avatar--contentStyle"
                    >
                      <span
                        class="bp3-popover-wrapper Avatar--avatarPopOver"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <div
                            class="Avatar--AvatarInner"
                            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
                            tabindex="0"
                          >
                            E
                          </div>
                        </span>
                      </span>
                    </div>
                    <div
                      class="Layout--vertical StyledProps--main StyledProps--padding-left-small"
                    >
                      <div
                        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                      >
                        <p
                          class="StyledProps--font Text--lineclamp StyledProps--main name StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                          style="--text-line-clamp: 1;"
                        >
                          example3
                        </p>
                        <p
                          class="StyledProps--font StyledProps--main colorBar pending StyledProps--font-size-xsmall StyledProps--font-weight-bold StyledProps--color StyledProps--color-primary7"
                        >
                          projectsOrgs.pendingInvitation
                        </p>
                      </div>
                      <p
                        class="StyledProps--font Text--lineclamp StyledProps--main email"
                        style="--text-line-clamp: 1;"
                      >
                        example3@email.com
                      </p>
                      <div
                        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                      >
                        <p
                          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                        >
                          projectsOrgs.roleAssigned
                        </p>
                        <p
                          class="StyledProps--font Text--lineclamp StyledProps--main role StyledProps--font-variation-tiny"
                          style="--text-line-clamp: 1;"
                        >
                          role 1
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Layout--horizontal StyledProps--main align toEnd StyledProps--padding-right-medium"
                    style="width: 40%;"
                  >
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--inline Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-repeat StyledProps--main"
                            icon="repeat"
                          >
                            <svg
                              data-icon="repeat"
                              height="16"
                              viewBox="0 0 16 16"
                              width="16"
                            >
                              <desc>
                                repeat
                              </desc>
                              <path
                                d="M10 5c0 .55.45 1 1 1h4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1s-1 .45-1 1v1.74A7.95 7.95 0 008 0C3.58 0 0 3.58 0 8c0 4.06 3.02 7.4 6.94 7.92.02 0 .04.01.06.01.33.04.66.07 1 .07 4.42 0 8-3.58 8-8 0-.55-.45-1-1-1s-1 .45-1 1c0 3.31-2.69 6-6 6-.71 0-1.37-.15-2-.38v.01C3.67 12.81 2 10.61 2 8c0-3.31 2.69-6 6-6 1.77 0 3.36.78 4.46 2H11c-.55 0-1 .45-1 1z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--inline Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-trash StyledProps--main"
                            icon="trash"
                          >
                            <svg
                              data-icon="trash"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <desc>
                                trash
                              </desc>
                              <path
                                d="M17 1h-5c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1H3c-.55 0-1 .45-1 1v1h16V2c0-.55-.45-1-1-1zm.5 3h-15c-.28 0-.5.22-.5.5s.22.5.5.5H3v14c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V5h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zM7 16c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8zm4 0c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8zm4 0c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="StyledProps--font StyledProps--main invites StyledProps--padding-left-xsmall StyledProps--padding-top-medium StyledProps--padding-bottom-medium"
              >
                <div
                  class="Layout--horizontal StyledProps--main"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main align pendingUser"
                    style="width: 60%;"
                  >
                    <div
                      class="Avatar--Avatar Avatar--contentStyle"
                    >
                      <span
                        class="bp3-popover-wrapper Avatar--avatarPopOver"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <div
                            class="Avatar--AvatarInner"
                            style="border-radius: 100%; line-height: 26px; width: 32px; height: 32px; font-size: 10px;"
                            tabindex="0"
                          >
                            E
                          </div>
                        </span>
                      </span>
                    </div>
                    <div
                      class="Layout--vertical StyledProps--main StyledProps--padding-left-small"
                    >
                      <div
                        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                      >
                        <p
                          class="StyledProps--font Text--lineclamp StyledProps--main name StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                          style="--text-line-clamp: 1;"
                        >
                          example2
                        </p>
                        <p
                          class="StyledProps--font StyledProps--main colorBar pending StyledProps--font-size-xsmall StyledProps--font-weight-bold StyledProps--color StyledProps--color-primary7"
                        >
                          projectsOrgs.pendingInvitation
                        </p>
                      </div>
                      <p
                        class="StyledProps--font Text--lineclamp StyledProps--main email"
                        style="--text-line-clamp: 1;"
                      >
                        example2@email.com
                      </p>
                      <div
                        class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                      >
                        <p
                          class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                        >
                          projectsOrgs.roleAssigned
                        </p>
                        <p
                          class="StyledProps--font Text--lineclamp StyledProps--main role StyledProps--font-variation-tiny"
                          style="--text-line-clamp: 1;"
                        >
                          role 1
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Layout--horizontal StyledProps--main align toEnd StyledProps--padding-right-medium"
                    style="width: 40%;"
                  >
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--inline Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-repeat StyledProps--main"
                            icon="repeat"
                          >
                            <svg
                              data-icon="repeat"
                              height="16"
                              viewBox="0 0 16 16"
                              width="16"
                            >
                              <desc>
                                repeat
                              </desc>
                              <path
                                d="M10 5c0 .55.45 1 1 1h4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1s-1 .45-1 1v1.74A7.95 7.95 0 008 0C3.58 0 0 3.58 0 8c0 4.06 3.02 7.4 6.94 7.92.02 0 .04.01.06.01.33.04.66.07 1 .07 4.42 0 8-3.58 8-8 0-.55-.45-1-1-1s-1 .45-1 1c0 3.31-2.69 6-6 6-.71 0-1.37-.15-2-.38v.01C3.67 12.81 2 10.61 2 8c0-3.31 2.69-6 6-6 1.77 0 3.36.78 4.46 2H11c-.55 0-1 .45-1 1z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--inline Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-trash StyledProps--main"
                            icon="trash"
                          >
                            <svg
                              data-icon="trash"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <desc>
                                trash
                              </desc>
                              <path
                                d="M17 1h-5c0-.55-.45-1-1-1H9c-.55 0-1 .45-1 1H3c-.55 0-1 .45-1 1v1h16V2c0-.55-.45-1-1-1zm.5 3h-15c-.28 0-.5.22-.5.5s.22.5.5.5H3v14c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V5h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zM7 16c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8zm4 0c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8zm4 0c0 .55-.45 1-1 1s-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal StyledProps--main"
        >
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main manageUsers Button--with-current-color Button--variation Button--variation-link"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              projectsOrgs.manageProject
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
